GDevelop.js/__tests__/GDJSBasicCodeGenerationIntegrationTests.js

```diff
@@ -190,31 +186,137 @@ describe('libGD.js - GDJS Code Generation integration tests', function () {
+  };
+  const falseEqualsCondition = {
+    type: { value: 'Egal' },
+    parameters: ['1', '=', '0'],
+  };
+
+  [
+    { a: false, b: false, result: false },
+    { a: false, b: true, result: false },
+    { a: true, b: false, result: false },
+    { a: true, b: true, result: true },
+  ].forEach(({ a, b, result }) => {
+    it(`can generate a condition list: ${a} and ${b} is ${result}`, function () {
+      const runtimeScene = generateAndRunVariableAffectationWithConditions([
+        a ? trueEqualsCondition : falseEqualsCondition,
+        b ? trueEqualsCondition : falseEqualsCondition,
+      ]);
+
+      expect(runtimeScene.getVariables().has('SuccessVariable')).toBe(result);
+      if (result) {
+        expect(
+          runtimeScene.getVariables().get('SuccessVariable').getAsNumber()
+        ).toBe(1);
+      }
+    });
+  });
+
+  [
+    { a: false, b: false, result: false },
+    { a: false, b: true, result: false },
+    { a: true, b: false, result: false },
+    { a: true, b: true, result: true },
+  ].forEach(({ a, b, result }) => {
+    it(`can generate a And: ${a} and ${b} is ${result}`, function () {
+      const runtimeScene = generateAndRunVariableAffectationWithConditions([
+        {
+          type: {
+            value: 'BuiltinCommonInstructions::And',
+          },
+          parameters: [],
+          subInstructions: [
+            a ? trueEqualsCondition : falseEqualsCondition,
+            b ? trueEqualsCondition : falseEqualsCondition,
+          ],
+        },
+      ]);
+
+      expect(runtimeScene.getVariables().has('SuccessVariable')).toBe(result);
+      if (result) {
+        expect(
+          runtimeScene.getVariables().get('SuccessVariable').getAsNumber()
+        ).toBe(1);
+      }
+    });
+  });
+
+  [
+    { a: false, b: false, result: true },
+    { a: false, b: true, result: true },
+    { a: true, b: false, result: true },
+    { a: true, b: true, result: false },
+  ].forEach(({ a, b, result }) => {
+    it(`can generate a Not: not(${a} and ${b}) is ${result}`, function () {
+      const runtimeScene = generateAndRunVariableAffectationWithConditions([
+        {
+          type: {
+            value: 'BuiltinCommonInstructions::Not',
+          },
+          parameters: [],
+          subInstructions: [
+            a ? trueEqualsCondition : falseEqualsCondition,
+            b ? trueEqualsCondition : falseEqualsCondition,
+          ],
+        },
+      ]);
+
+      expect(runtimeScene.getVariables().has('SuccessVariable')).toBe(result);
+      if (result) {
+        expect(
+          runtimeScene.getVariables().get('SuccessVariable').getAsNumber()
+        ).toBe(1);
+      }
+    });
+  });
+
+  [
+    { a: false, b: false, result: false },
+    { a: false, b: true, result: true },
+    { a: true, b: false, result: true },
+    { a: true, b: true, result: true },
+  ].forEach(({ a, b, result }) => {
+    it(`can generate a Or: ${a} or ${b} is ${result}`, function () {
```

Comment:
Could you add a test case for:
- nested Or/And/Not
- one or more "stress" cases of a And containing multiple Ors containing multiple Nots (or vice versa)?

The goal in both case is to validate that the nesting is properly done.